labs/a8.md

```diff
@@ -0,0 +1,736 @@
+owner.
+5. `file6.txt` and `file7.txt`: look at the permissions for these two files and 
+explain what they allow and disallow.
+6. `file8.txt` and `file9.txt`: provide a strategy to make these files readable 
+only to you and the lab8user, and no one else. 
+
+### Network Security 
+
+On UNIX based operating systems, the network is the most common method of 
+gaining unauthorized access to a machine. Why this is the case should be 
+obvious: the network allows one to interact with a machine without needing any 
+physical presence. Anyone on the same network as your machine can connect, and 
+therefore attack, your machine. On the internet, this means the whole world can 
+attack your machine. This introduces a number of security considerations when 
+running networked applications.
+
+The first step in securing machines connected to a network, or the open 
+internet, is to make it as hard as possible for attackers to get to your 
+machine. As you may recall from the networking lecture, processes interact with 
+one another over a network by means of sockets and ports. Many protocols listen 
+on well-known ports, such as SSH on port 22, HTTP on port 80, FTP on port 21, 
+etc. We use firewalls and similar software to prevent unauthorized users from 
+connecting to your machine. For example, the student VMs are heavily firewalled 
+- the only traffic allowed from the internet at large is SSH (TCP to port 22), 
+HTTP/S (TCP to port 80, 443, 8000, 8080) and a few other services necessary for 
+labs (e.g. UDP to port 53 for lab 7).
+
+The second step in maintaining network security is to audit which programs are 
+running on your server, and to design your network to reduce the attack vectors 
+on any single machine by separating concerns as much as reasonably possible. 
+For example, it may not be a good idea to run a critical database server and a 
+potentially insecure webserver on the same machine, as a vulnerability in one 
+could easily lead to the compromising of the other. 
+
+Finally, the hardest part of online security is auditing your own code. 
+Ultimately, most security vulnerabilities arise from code you need to run to 
+affect your business functions. For example, the popular CMS software 
+WordPress, which is estimated to power nearly a quarter of all websites, has
+historically been extremely vulnerable to security bugs. Moreover, people like 
+to install WordPress plugins, often written by amateur coders, which are even 
+more insecure. Unfortunately, aside from being vigilant about monitoring your 
+servers for intrusion and generally following good defensive programming 
+practices, there isn't much one can do to avoid bugs in their own code. 
+
+
+### Network Security Lab Activity
+
+1. Make a list of all the services running on your student VM that are 
+accessible from the public internet, what user they are running as, and what 
+port they are listening on.
```

Comment:
Usually I'm in favor of less explanation, but in this case I think it could be valuable to include a nudge in the right direction (unless it can be found in the slides?)

Reply:
This question is deceptively hard. Really the best way to determine what is truly accessible from the public internet is to scan the VM from a machine outside of the subnet, but we probably don't want them to do this.